teamwork/apps/courses/views/CourseView.py

```diff
@@ -175,13 +175,18 @@ def update_course(request, slug):
     page_description = "Update %s"%(course.name) or "Post a new update"
     title = "Update Course"
 
+    if not request.user.profile.isGT:
+        user_role = Enrollment.objects.filter(user=request.user, course=course).first().role
+    else:
+        user_role = 'GT'
+
     if request.user.profile.isGT:
         pass
-    #if user is not a professor or they did not create course
-    elif not course.creator == request.user:
-        #redirect them to the /course directory with message
-        messages.info(request,'Only Professor can post and update')
-        return HttpResponseRedirect('/course')
+    elif not request.user==course.creator:
```

Comment:
elif user_role is not equal to 'professor' or 'ta'

Reply:
This does the same thing

Reply:
No shit. One line of code versus two though